R/build_site.R

```diff
@@ -15,14 +15,14 @@
 #' json <- write_json(biblio, access, attributes, creators)
 #' build_site(json)
 #' }
-build_site <- function(path = "data/metadata/dataspice.json",
+build_site <- function(path = file.path("data", "metadata", "dataspice.json"),
                        template_path = system.file("template.html5",
                                                    package = "dataspice"),
-                       out_path = "docs/index.html") {
+                       out_path = file.path("docs", "index.html")) {
   data <- jsonld_to_mustache(path)
 
     # Make docs dir if not present
-  if (!file.exists("docs")) {
+  if (dirname(out_path) == 'docs' & !file.exists("docs")) {
```

Comment:
I think we put a bit of a logic bug in here. The idea is that whatever is given in the `out_path` argument will be used. Your changes improves the checking I think it can go one step farther:

```r
# Make docs dir if not present
out_dir <- dirname(out_path)

if (!dir.exists(out_dir) {
  dir.create(out_dir, recursive = TRUE)
}
```

This allows any `out_path`, even paths a few levels deep. What do you think?